linker-interface/shared/src/main/scala/org/scalajs/linker/interface/StandardConfig.scala

```diff
@@ -239,6 +269,100 @@ object StandardConfig {
    */
   def apply(): StandardConfig = new StandardConfig()
 
+  /** Tests whether a string is a valid JS header.
+   *
+   *  A header is valid if and only if it satisfies the following constraints:
+   *
+   *  - It must contain only valid JS whitespace and/or JS comments (single- or multi-line).
+   *  - It must not use new line characters that are not UNIX new lines (`"\n"`).
+   *  - If non-empty, it must end with a new line.
+   *  - It must not contain unpaired surrogate characters (i.e., it must be a valid UTF-16 string).
+   */
+  def isValidJSHeader(jsHeader: String): Boolean = {
```

Comment:
My main question / concern is with the evolution / generality of this method. Mainly: Is it imaginable that its specification needs to change and/or depend on linker specifics? (e.g. a new way of having comments in JS, unicode std changes, ?).

Or in other words: Are we OK with restricting linkers to absolutely conform to this? If yes, this is the right place. If no, we might need to move this into the Linker (and live with the worse usability of rejecting a bad config later).

Reply:
JavaScript is very careful about its backwards compatibility. So I'm quite confident that anything that is a valid JavaScript whitespace/comment will stay a valid JavaScript whitespace/comment forever. Regarding Unicode, the set of `\p{Zs}` can only grow over time.

Regarding newline characters, the Unicode standard says that, in theory, it could grow. See https://www.unicode.org/reports/tr14/tr14-32.html#Properties, section "Future updates". We could forbid Unassigned characters (category Cn) but that still does not fully protect us against a *change* of Line Breaking Property for an existing character. That seems like a very unlikely scenario, however, and if it happens, well, I think we can live with the future breaking (or with accepting the newly-characterized line breaking code points).

Reply:
Hmmm... OK. What about allowing `#` as a comment character (just for example ;))? Wouldn't we need to know the output target to detect what would be allowed?

Reply:
(so just TBC, I'm not concerned about backwards compat, I'm concerned about forwards compat, basically that we'll be forced to be too restricitve)

Reply:
Ah, forward compatibility! Indeed. Well, at least it hasn't changed since ECMAScript 5.1: https://262.ecma-international.org/5.1/#sec-7.2. That doesn't mean that it can't change in a future version, though.

I think it's not really a bid deal, here. What we really need to check is that `jsHeader` cannot be used to inject actual code. If the user injects something that causes a `ParseError` in the ES version that they target, it's probably fine to say that it's on them. But if they inject something that *does* parse and result in executable code, that is a problem, and we need to prevent that.

Reply:
Fair enough. And it will not happen that something that is a comment in a later version is code in an earlier version. So we're all good :)